### PR TITLE
New game UI raid

### PR DESCRIPTION
**Pull request by Spintacle123**
PR Includes:
 - New UI for Raid
 - Sidebar Indicator if Character is on Raid
 - Sidebar Indicator if Character in on PVP
 - Added Scroll on modal menu in weapon selection when in mobile view

### CODE REVIEW

File: frontend/src/locales/en.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "English",
   "viewLink": {
     "plaza": "Plaza",
-    "character" : "Character",
+    "viewLink" : "Character",
```
Comment (kevinunger) [inline frontend/src/locales/en.json:5]: why is that changed?

File: frontend/src/components/WeaponInvetory.vue [hunk 2]
```diff
@@ -328,12 +388,15 @@ export default {
   min-width: 18px;
 }
 
+.raid-weapon{
+  font-size: 0.7em !important;
```
Comment (kevinunger) [inline frontend/src/components/WeaponInvetory.vue:392]: do we need the !important here? We should avoid those if possible

**Comment by kevinunger** [comment]
you commited the package-lock. Please revert that. You can also downgrade your npm to version 6. Then you'll have the lock file in version 1 as well (npm install -g npm@6)

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 3]
```diff
@@ -109,6 +154,53 @@ export default Vue.extend({
       else return 'character';
     },
 
+    //getNftStatus(chars: any): string {
```
Comment (kevinunger) [inline frontend/src/components/smart/CharacterDisplay.vue:157]: remove please

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 4]
```diff
@@ -109,6 +154,53 @@ export default Vue.extend({
...
+    getNftStatus(chars: any){
+      // const charStatus = await this.composeCharacterData(chars.id);
```
Comment (kevinunger) [inline frontend/src/components/smart/CharacterDisplay.vue:159]: remove please

File: frontend/src/views/Raid.vue [hunk 5]
```diff
@@ -273,6 +474,7 @@ const dragonNames = [
   // 'Richard Melics',
 ];
 
+// const moment = require('moment');
```
Comment (kevinunger) [inline frontend/src/views/Raid.vue:477]: if this is unused, can it be removed from the package.json and lockfile?

File: frontend/src/views/Raid.vue [hunk 6]
```diff
@@ -609,27 +887,37 @@ export default Vue.extend({
     interval = window.setInterval(async () => {
       await refreshRaidData();
     }, 3000);
+
+    // (this as any).$bvModal.show('rewardsModal');
```
Comment (kevinunger) [inline frontend/src/views/Raid.vue:891]: remove

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 7]
```diff
@@ -109,6 +156,51 @@ export default Vue.extend({
       else return 'character';
     },
 
+    getNftStatus(chars: any){
+      this.composeCharacterData(chars.id).then(data => {
+        if (data.status !== undefined && data.status in NftStatus) {
+          for(const a of this.filteredCharactersForList){
+            if(a.id === chars.id){
```
Comment (Sellie96) [inline frontend/src/components/smart/CharacterDisplay.vue:163]: Name the variable properly

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 8]
```diff
@@ -109,6 +156,51 @@ export default Vue.extend({
...
+              a.pvpStatus = NftStatus[data.status];
+            }
+          }
+        } else {
+          for(const a of this.filteredCharactersForList){
```
Comment (Sellie96) [inline frontend/src/components/smart/CharacterDisplay.vue:168]: Again

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 9]
```diff
@@ -109,6 +156,51 @@ export default Vue.extend({
...
+      const charId = id;
+      if(await this.isCharacterAlreadyRaiding(id)){
+        this.charOnRaid.push(charId);
+      }
+    },
+
+    getIfCharacterIsInRaid(id: any){
+      let toReturnWarning;
+      this.charOnRaid.forEach(y => {
+        if(id === y){
```
Comment (Sellie96) [inline frontend/src/components/smart/CharacterDisplay.vue:196]: Name not letter

File: frontend/src/views/Raid.vue [hunk 10]
```diff
@@ -1,218 +1,411 @@
...
+        <p>{{$t('raid.raidRewardsSelector')}}</p>
+        <p>{{$t('raid.selectRaid')}}</p>
+        <img src="../assets/separator.png" alt="">
+     </div>
+      <div class="raid-picker" v-if="!isLoading">
+        <div class="raid-rewards">
+          <div v-for="id in rewardIndexes" :value="id" :key="id" @click="closeRewardPicker(id)">
+            <p class="m-0">Raid # {{id}}</p>
+            <img src="../assets/chest.png" alt="Chest Icon">
+            <button class="btn-rewards">CLAIM</button>
```
Comment (Sellie96) [inline frontend/src/views/Raid.vue:385]: translaiton required

File: frontend/src/views/Raid.vue [hunk 11]
```diff
@@ -1,218 +1,411 @@
...
+                        <div class="sec">
+                          <p>{{ zeroPad(remainingTime.seconds, 2)}}</p>
+                          <span>{{ remainingTime.seconds > 1 ? $t('raid.sec') : $t('raid.sec')}}</span>
+                        </div>
+                    </div>
+                  </div>
+
+                  <!-- HIDE THIS FOR NOW THIS DESCRIPTION OF THE BOSS-->
+                  <!-- <div class="boss-history">
+                    Messenger ravens flood the skies with news of the Seers visions. Heralds are heard throughout the cities and towns proclaiming.
```
Comment (Sellie96) [inline frontend/src/views/Raid.vue:76]: Hiding why?

File: frontend/src/locales/en.json [hunk 12]
```diff
@@ -459,7 +486,7 @@
     "loading": "Loading...",
     "gotXP": "All participating characters got {xpReward} XP",
     "gotBonusXP": "Following character(s) performed exceptionally well and were rewarded with bonus XP:",
-    "loot": "Loot",
+    "loot": "Raid Loots",
```
Comment (Sellie96) [inline frontend/src/locales/en.json:489]: Revert this or drop the s to "Raid Loot"

File: frontend/src/components/smart/WeaponRowGrid.vue [hunk 13]
```diff
@@ -410,12 +410,12 @@ export default Vue.extend({
       this.maxPriceFilter= '';
       this.$emit('weapon-filters-changed');
     },
-    onWeaponClick(id: number) {
-      this.setCurrentWeapon(id);
-      this.$emit('chooseweapon', id);
-      this.$emit('choose-weapon', id);
+    onWeaponClick(weapon: any) {
```
Comment (Sellie96) [inline frontend/src/components/smart/WeaponRowGrid.vue:413]: Can weapon be a defined type?
